cordyceps/src/sorted_list.rs

```diff
@@ -0,0 +1,459 @@
+//! [Intrusive], singly-linked, sorted, linked-list.
+//!
+//! See the documentation for the [`SortedList`] and [`SortedListIter`] types for
+//! details.
+//!
+//! [intrusive]: crate#intrusive-data-structures
+
+use crate::{Linked, Stack};
+use core::{marker::PhantomData, ptr::NonNull};
+
+pub use crate::stack::Links;
+
+/// A sorted singly linked list
+///
+/// This behaves similar to [`Stack`], in that it is a singly linked list,
+/// however items are stored in an ordered fashion. This means that insertion
+/// is an `O(n)` operation, and retrieval of the first item is an `O(1)` operation.
+///
+/// It allows for a user selected ordering operation. If your type `T` implements
+/// [`Ord`]:
+///
+/// * Consider using [`SortedList::new_min()`] if you want **smallest** items sorted first.
+/// * Consider using [`SortedList::new_max()`] if you want **largest** items sorted first.
+///
+/// If your type `T` does NOT implement [`Ord`], or you want to use
+/// a custom sorting anyway, consider using [`SortedList::new_custom()`]
+#[derive(Debug)]
+pub struct SortedList<T: Linked<Links<T>>> {
+    head: Option<NonNull<T>>,
+    // Returns if LHS is less/same/greater than RHS
+    func: fn(&T, &T) -> core::cmp::Ordering,
+}
+
+#[inline]
+fn invert_sort<T: core::cmp::Ord>(a: &T, b: &T) -> core::cmp::Ordering {
+    // Inverted sort order!
+    T::cmp(b, a)
```

Comment:
I was wondering why `core:;cmp` didn't already have something like this baked in, and I guess that's because it has [this instead](https://doc.rust-lang.org/stable/std/cmp/struct.Reverse.html), which is a bit annoying to use here...

Reply:
Ah interesting! I should have gone looking, but yeah I think the current solution is reasonable.